loading.bs

```diff
@@ -331,6 +385,376 @@ Note: Applying the signed exchange's response here has the effect of letting a
+        1. Run the following steps [=in parallel=]:
+            1. Let |sxgResponse| be the result of [=fetching=] |sxgRequest|.
+            1. If |sxgResponse|'s [=response/came from a signed exchange=] is
+                true, and |sxgResponse|'s [=response/URL=] is the same as
+                |linkTarget|, then create an [=exchange=] with the URL of
+                |linkTarget| and the result of [=response/cloning=]
+                |sxgResponse|, and store the exchange to |el|'s [=Node/node
+                document=]'s [=Document/prefetched subresource signed
+                exchanges=].
+
+
+Issue(w3c/resource-hints#77): Currently the behavior of recursive prefetch is
+not defined in the specs. This section monkeypatches Link type
+"<{link/rel/prefetch}>" to support the recursive prefetching of alternate signed
+exchange subresources. When we will change the HTML spec to support the general
+recursive prefetching, this section must be made align with the change.
+
+<h3 algorithm id="mp-process-a-navigate-fetch">
+Monkeypatch process a navigate fetch</h3>
+In [process a navigate fetch](https://html.spec.whatwg.org/multipage/browsing-the-web.html#process-a-navigate-fetch)
+before
+
+> 9. While true:
+
+add the following steps:
+
+    9. If |sourceBrowsingContext|'s [=active document=]'s
+        [=Document/prefetched signed exchanges for navigation=][|request|'s
+        [=request/url=]] [=map/exists=], then:
+
+        1. Let |prefetchedExchange| be |sourceBrowsingContext|'s 
+            [=active document=]'s [=Document/prefetched signed exchanges for
+            navigation=][|request|'s [=request/url=]].
+
+        1. Set |request|'s [=request/stashed exchange=] to the result of
+            creating a new [=exchange=] with |prefetchedExchange|'s
+            [=exchange/request URL=] and the result of [=response/cloning=]
+            |prefetchedExchange|'s [=exchange/response=].
+
+
+And before
+
+> 17. Run process a navigate response with navigationType, the source browsing
+>     context, and navigationParams.
+
+add the following steps:
+
+    18. Copy |sourceBrowsingContext|'s [=active document=]'s
+        [=Document/prefetched subresource signed exchanges=]
+        to <var ignore=''>navigationParams</var>'s
+        [=navigation params/prefetched subresource signed exchanges=].
+
+Note: As browsers move toward partitioned HTTP caches, the source document's
+    cache will likely be separate from the target's cache, so we can't just pass
+    prefetched content through the cache.
+
+
+<h3 algorithm id="mp-page-load-processing-model-for-html-fiiles">
+Monkeypatch Page load processing model for HTML files</h3>
+In
+[Page load processing model for HTML files](https://html.spec.whatwg.org/multipage/browsing-the-web.html#read-html)
+before
+
+> 2. Create an HTML parser and ...
+
+add the following steps:
+    2. Run the following steps [=in parallel=]:
+        1. Wait until |document|'s [=viewport=] is initialized.
```

Comment:
I don't think that's what you want to do here. You want to wait until the viewport is "known" (in practice, by scanning HTML tokens). Otherwise, IIUC, the viewport is initialized by the parser, which creation you're blocking here...
Might be worthwhile to add a note/TODO saying that this needs to be better defined if/when we'll define the preloadScanner

Reply:
Added Note.